### PR TITLE
Adding dartdocs to flame_svg

### PR DESCRIPTION
**Pull request by erickzanardo**
# Description

First PR of a series to add dartdoc on all of our code base. Starting with the `flame_svg`.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors and are passing (See [Contributor Guide]).
- [x] My PR does not decrease the code coverage, or I have __a very special case__ and explained on the PR description why this PR decreases the coverage.
- [ ] I updated/added relevant documentation (doc comments with `///`) and updated/added examples in `doc/examples`.
- [x] I have formatted my code with `flutter format` and the `flutter analyze` does not report any problems.
- [x] I read and followed the [Flame Style Guide].
- [ ] I have added a description of the change under `[next]` in `CHANGELOG.md`.
- [x] I removed the `Draft` status, by clicking on the `Ready for review` button in this PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flame users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in `CHANGELOG.md`).
- [x] No, this is *not* a breaking change.

## Related Issues

*Replace this paragraph with a list of issues related to this PR from the [issue database]. Indicate, which of these issues are resolved or fixed by this PR.*

[issue database]: https://github.com/flame-engine/flame/issues
[Contributor Guide]: https://github.com/flame-engine/flame/blob/main/CONTRIBUTING.md
[Flame Style Guide]: https://github.com/flame-engine/flame/blob/main/STYLEGUIDE.md

### CODE REVIEW

File: packages/flame_svg/lib/svg.dart [hunk 1]
```diff
@@ -36,6 +41,8 @@ class Svg {
   }
 }
 
+/// Provides loading extensions for [Svg] on [Game] class.
```
Comment (spydon) [inline packages/flame_svg/lib/svg.dart:44]:
```suggestion
/// Provides a loading method for [Svg] on the [Game] class.
```

File: packages/flame_svg/lib/svg.dart [hunk 2]
```diff
@@ -36,6 +41,8 @@ class Svg {
...
 extension SvgLoader on Game {
+  /// Loads a [Svg] using [Game] own asset loader.
```
Comment (spydon) [inline packages/flame_svg/lib/svg.dart:46]:
```suggestion
  /// Loads a [Svg] using the [Game]'s own asset loader.
```

File: packages/flame_svg/lib/svg_component.dart [hunk 3]
```diff
@@ -4,9 +4,12 @@ import 'package:flame/components.dart';
 
 import './svg.dart';
 
+/// Wraps a [Svg] into a Flame component.
 class SvgComponent extends PositionComponent {
+  /// [Svg] instance
```
Comment (spydon) [inline packages/flame_svg/lib/svg_component.dart:9]: These type of docs are the ones I'm afraid we will get too many of when we get in to the Flame codebase.

File: packages/flame_svg/lib/svg_component.dart [hunk 4]
```diff
@@ -4,9 +4,12 @@ import 'package:flame/components.dart';
 
 import './svg.dart';
 
+/// Wraps a [Svg] into a Flame component.
```
Comment (spydon) [inline packages/flame_svg/lib/svg_component.dart:7]:
```suggestion
/// Wraps [Svg] in a Flame component.
```

File: packages/flame_svg/lib/svg_component.dart (hunk 3 above)
Comment (erickzanardo) [reply packages/flame_svg/lib/svg_component.dart:9]: Yep, but the pros of the other docs outweight the cons of these useless comments IMO :D

**Review by spydon — APPROVED** [review]

File: packages/flame_svg/lib/svg_component.dart (hunk 3 above)
Comment (spydon) [reply packages/flame_svg/lib/svg_component.dart:9]: You're probably right!

Comment (st-pasha) [reply packages/flame_svg/lib/svg_component.dart:9]: Another possibility is to make it private

Comment (luanpotter) [reply packages/flame_svg/lib/svg_component.dart:9]: how about `The wrapped instance of [Svg].` (given that the class doc says it wraps Svg into a component)

File: packages/flame_svg/lib/svg_component.dart [hunk 5]
```diff
@@ -4,9 +4,12 @@ import 'package:flame/components.dart';
 
 import './svg.dart';
 
+/// Wraps [Svg] in a Flame component.
 class SvgComponent extends PositionComponent {
+  /// [Svg] instance
   Svg svg;
 
+  /// Creates a [SvgComponent] from a [Svg] instance.
```
Comment (luanpotter) [inline packages/flame_svg/lib/svg_component.dart:12]:
```suggestion
  /// Creates an [SvgComponent] from an [Svg] instance.
```

File: packages/flame_svg/lib/svg.dart [hunk 6]
```diff
@@ -36,6 +41,8 @@ class Svg {
   }
 }
 
+/// Provides a loading method for [Svg] on the [Game] class.
 extension SvgLoader on Game {
+  /// Loads a [Svg] using the [Game]'s own asset loader.
```
Comment (luanpotter) [inline packages/flame_svg/lib/svg.dart:46]:
```suggestion
  /// Loads an [Svg] using the [Game]'s own asset loader.
```

File: packages/flame_svg/lib/svg.dart [hunk 7]
```diff
@@ -8,11 +8,16 @@ import 'package:flame/game.dart';
...
 
+/// A [Svg] to be rendered on a Flame [Game].
 class Svg {
+  /// The [DrawableRoot] that this [Svg] represents.
   DrawableRoot svgRoot;
 
+  /// Creates a [Svg] with the received [svgRoot].
   Svg(this.svgRoot);
 
+  /// Loads a [Svg] with the received [cache]. When no [cache] is provided, the global
```
Comment (luanpotter) [inline packages/flame_svg/lib/svg.dart:19]:
```suggestion
  /// Loads an [Svg] with the received [cache]. When no [cache] is provided, the global
```

File: packages/flame_svg/lib/svg.dart [hunk 8]
```diff
@@ -8,11 +8,16 @@ import 'package:flame/game.dart';
 import 'package:flutter/material.dart';
 import 'package:flutter_svg/flutter_svg.dart';
 
+/// A [Svg] to be rendered on a Flame [Game].
 class Svg {
+  /// The [DrawableRoot] that this [Svg] represents.
   DrawableRoot svgRoot;
 
+  /// Creates a [Svg] with the received [svgRoot].
```
Comment (luanpotter) [inline packages/flame_svg/lib/svg.dart:16]:
```suggestion
  /// Creates an [Svg] with the received [svgRoot].
```

**Review by luanpotter — APPROVED** [review]
LGTM

File: packages/flame_svg/lib/svg_component.dart (hunk 3 above)
Comment (erickzanardo) [reply packages/flame_svg/lib/svg_component.dart:9]:
```suggestion
  /// The wrapped instance of [Svg].
```

**Pull request merged**